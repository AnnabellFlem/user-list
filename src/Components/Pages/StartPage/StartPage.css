@import '../../../Styles/colors.css';
@import '../../../Styles/sizes.css';
@import '../../../Styles/start-page-base.css';

.main {
  background-image: linear-gradient(-38deg, var(--orange-5) 50%, var(--white) 50%);
}

@media (max-width: 1200px) {
  .main {
    background-image: linear-gradient(-37deg, var(--orange-5) 50%, var(--white) 50%);
  }
}

@media (max-width: 920px) {
  .main {
    background-image: linear-gradient(-41deg, var(--orange-5) 50%, var(--white) 50%);
  }
}

@media (max-width: 600px) {
  .main {
    background-image: linear-gradient(-53deg, var(--orange-5) 50%, var(--white) 50%);
  }
}

@media (max-width: 450px) {
  .main {
    background-image: linear-gradient(-61deg, var(--orange-5) 50%, var(--white) 50%);
  }
}

@media (max-width: 360px) {
  .main {
    background-image: linear-gradient(-66deg, var(--orange-5) 50%, var(--white) 50%);
  }
}

@media (max-width: 320px) {
  .main {
    background-image: linear-gradient(-68deg, var(--orange-5) 50%, var(--white) 50%);
  }
}
