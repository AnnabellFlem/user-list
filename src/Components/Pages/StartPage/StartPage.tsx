import React from 'react'
import './StartPage.css'
import Button from '../../BaseComponents/Button'
import { ReactComponent as ThumbsUpLogo } from '../../../Images/svg/thumbs-up.svg'
import useMedia from 'use-media'
import { MEDIA_QUERY_MOBILE } from '../../../Constants/MediaQueries'

type Props = Partial<{
  onButtonClick: (e: React.MouseEvent<HTMLButtonElement | HTMLElement>) => void
}>

const StartPage: React.FC<Props> = ({ onButtonClick }) => {
  const isWide = useMedia(MEDIA_QUERY_MOBILE)

  return (
    <main>
      <section className="main">
        <div className="main__wrapper">
          <div className="main__image">
            <ThumbsUpLogo />
          </div>
          <div className="main__text">
            <h1 className="main__title">Who wants to be a millionaire?</h1>
            {isWide ? (
              <Button onClick={onButtonClick} value="start">
                Start
              </Button>
            ) : (
              <Button onClick={onButtonClick} size="s" fullWidth value="start">
                Start
              </Button>
            )}
          </div>
        </div>
      </section>
    </main>
  )
}

export default StartPage
