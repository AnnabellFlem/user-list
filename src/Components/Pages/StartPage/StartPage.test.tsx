import React from 'react'
import { shallow } from 'enzyme'
import StartPage from './index'

describe('StartPage test', function () {
  let wrapper: any

  beforeAll(() => {
    wrapper = shallow(<StartPage />)
  })

  it('should to match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should render component', () => {
    expect(wrapper.find('main')).toBeDefined()
    expect(wrapper.find('main')).toHaveLength(1)
  })
})
