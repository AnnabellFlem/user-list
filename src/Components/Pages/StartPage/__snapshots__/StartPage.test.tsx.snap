// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartPage test should to match snapshot 1`] = `ShallowWrapper {}`;
